### PR TITLE
Coverage Fix

### PR DESCRIPTION
**Pull request by allenwq**
Sorry, missing some specs before

### CODE REVIEW

File: spec/controllers/course/groups_controller_spec.rb [hunk 1]
```diff
@@ -39,6 +40,22 @@
...
+            first_id = generate(:nested_attribute_new_id)
+            second_id = generate(:nested_attribute_new_id)
+            {
+              first_id => attributes_for(:course_group_user, user_id: user_to_add),
+              second_id => attributes_for(:course_group_user, user_id: user_to_add)
+            }
+          end
+
+          it 'adds neither of them to the group' do
+            expect { subject }.to change { group.users.count }.by(0)
```
Comment (lowjoel) [inline spec/controllers/course/groups_controller_spec.rb]: maybe you should check that the error is also added?

File: spec/controllers/user/emails_controller_spec.rb [hunk 2]
```diff
@@ -16,6 +16,18 @@
         end
         it { is_expected.to redirect_to(user_emails_path) }
       end
+
+      context 'when destroy a primary email' do
```
Comment (lowjoel) [inline spec/controllers/user/emails_controller_spec.rb]: when destroying

File: spec/controllers/user/emails_controller_spec.rb [hunk 3]
```diff
@@ -8,7 +8,7 @@
     before { sign_in(user) }
 
     describe '#destroy' do
-      subject { delete :destroy, id: user.emails.first }
+      subject { delete :destroy, id: user.send(:default_email_record) }
 
       context 'when user only have one email' do
```
Comment (lowjoel) [inline spec/controllers/user/emails_controller_spec.rb]: when the user only has one email address

File: spec/controllers/course/groups_controller_spec.rb (hunk 1 above)
Comment (lowjoel) [inline spec/controllers/course/groups_controller_spec.rb]: @allenwq do you want to check the error?

Comment (allenwq) [inline spec/controllers/course/groups_controller_spec.rb]: Oh, I've added one more `it` below which checks the errors.

Comment (lowjoel) [inline spec/controllers/course/groups_controller_spec.rb]: ...where?

Comment (allenwq) [inline spec/controllers/course/groups_controller_spec.rb]: ... I'm sorry, but something went wrong ><

Comment (lowjoel) [inline spec/controllers/course/groups_controller_spec.rb]: sounds like the android app force close message :stuck_out_tongue:

File: spec/controllers/course/groups_controller_spec.rb [hunk 4]
```diff
@@ -39,6 +40,29 @@
...
+          end
+
+          it 'adds errors to group users' do
+            subject
+            user_with_errors = controller.instance_variable_get(:@group).group_users.
+                                          select { |user| user.errors.any? }
```
Comment (lowjoel) [inline spec/controllers/course/groups_controller_spec.rb]: @allenwq check using rubocop locally.

File: spec/controllers/course/groups_controller_spec.rb [hunk 5]
```diff
@@ -39,6 +40,29 @@
...
+            expect(user_with_errors).not_to be_empty
```
Comment (lowjoel) [inline spec/controllers/course/groups_controller_spec.rb]: you can check the error message also, actually.

File: spec/controllers/course/groups_controller_spec.rb (hunk 4 above)
Comment (allenwq) [inline spec/controllers/course/groups_controller_spec.rb]: My rubocop does not have such complain.

File: spec/controllers/course/groups_controller_spec.rb (hunk 5 above)
Comment (allenwq) [inline spec/controllers/course/groups_controller_spec.rb]: Sure

File: spec/controllers/course/groups_controller_spec.rb (hunk 4 above)
Comment (lowjoel) [inline spec/controllers/course/groups_controller_spec.rb]: then ignore.

this line should also be find, not select.

File: spec/controllers/course/groups_controller_spec.rb [hunk 6]
```diff
@@ -39,6 +40,32 @@
...
+            user_with_errors.each do |user|
+              expect(user.errors.messages[:user]).to include(I18n.t('errors.messages.taken'))
```
Comment (lowjoel) [inline spec/controllers/course/groups_controller_spec.rb]: then there's no need to use each.

File: spec/controllers/course/groups_controller_spec.rb (hunk 4 above)
Comment (allenwq) [inline spec/controllers/course/groups_controller_spec.rb]: The errors are added to all the duplicate group_users, so should be select

Comment (lowjoel) [inline spec/controllers/course/groups_controller_spec.rb]: right, i get you.

Comment (allenwq) [inline spec/controllers/course/groups_controller_spec.rb]: Thanks !

**Pull request merged**